From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: jmp <jasonpenilla2@me.com>
Date: Sat, 8 May 2021 19:36:30 -0700
Subject: [PATCH] temp fix compile


diff --git a/src/main/java/com/destroystokyo/paper/event/server/AsyncTabCompleteEvent.java b/src/main/java/com/destroystokyo/paper/event/server/AsyncTabCompleteEvent.java
index 31c90ab0710020a2ff104b5cb8af604f3d682f31..aad8d20c1f3433dba7036ae3ccf8e68c5dff2319 100644
--- a/src/main/java/com/destroystokyo/paper/event/server/AsyncTabCompleteEvent.java
+++ b/src/main/java/com/destroystokyo/paper/event/server/AsyncTabCompleteEvent.java
@@ -40,6 +40,8 @@ import org.bukkit.event.HandlerList;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.stream.Stream;
+
+import org.checkerframework.checker.nullness.qual.NonNull;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
@@ -136,7 +138,7 @@ public class AsyncTabCompleteEvent extends Event implements Cancellable {
      *
      * @return a list of offered completions
      */
-    public @NotNull List<@NotNull Completion> completions() {
+    public @NotNull List<@NonNull Completion> completions() {
         return this.completions;
     }
 
@@ -150,7 +152,7 @@ public class AsyncTabCompleteEvent extends Event implements Cancellable {
      *
      * @param newCompletions the new completions
      */
-    public void completions(final @NotNull List<@NotNull Completion> newCompletions) {
+    public void completions(final @NotNull List<@NonNull Completion> newCompletions) {
         Validate.notNull(newCompletions, "new completions");
         this.completions.clear();
         this.completions.addAll(newCompletions);
