From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: jmp <jasonpenilla2@me.com>
Date: Mon, 10 May 2021 21:16:01 -0700
Subject: [PATCH] Fix color of certain log messages


diff --git a/src/main/java/net/minecraft/server/MinecraftServer.java b/src/main/java/net/minecraft/server/MinecraftServer.java
index cca6d3f836b59717d2c9c6f251ee930026b338e7..3b23df9c87dfbe6196d01d7cb7e6a6905b52b585 100644
--- a/src/main/java/net/minecraft/server/MinecraftServer.java
+++ b/src/main/java/net/minecraft/server/MinecraftServer.java
@@ -12,7 +12,9 @@ import com.mojang.datafixers.DataFixer;
 import io.netty.buffer.ByteBuf;
 import io.netty.buffer.ByteBufOutputStream;
 import io.netty.buffer.Unpooled;
+import io.papermc.paper.adventure.PaperAdventure;
 import io.papermc.paper.event.entity.EntityMoveEvent;
+import io.papermc.paper.text.PaperComponents;
 import io.papermc.paper.util.TraceUtil;
 import it.unimi.dsi.fastutil.longs.LongIterator;
 import java.awt.image.BufferedImage;
@@ -1688,7 +1690,7 @@ public abstract class MinecraftServer extends IAsyncTaskHandlerReentrant<TickTas
 
     @Override
     public void sendMessage(IChatBaseComponent ichatbasecomponent, UUID uuid) {
-        MinecraftServer.LOGGER.info(org.bukkit.craftbukkit.util.CraftChatMessage.fromComponent(ichatbasecomponent));// Paper - Log message with colors
+        MinecraftServer.LOGGER.info(PaperComponents.legacySectionSerializer().serialize(PaperAdventure.asAdventure(ichatbasecomponent))); // Paper - Log message with colors
     }
 
     public KeyPair getKeyPair() {
