From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: jmp <jasonpenilla2@me.com>
Date: Sat, 21 Nov 2020 00:41:55 -0800
Subject: [PATCH] Remove server gui


diff --git a/src/main/java/net/minecraft/server/Main.java b/src/main/java/net/minecraft/server/Main.java
index 15b972b4a93b8fe3655acec47bc84b0f2c4620a6..63895731234d19cfc2ef1ede3098f9f5a5364ff8 100644
--- a/src/main/java/net/minecraft/server/Main.java
+++ b/src/main/java/net/minecraft/server/Main.java
@@ -236,7 +236,7 @@ public class Main {
 
                 if(!Boolean.parseBoolean(System.getenv().getOrDefault("PAPER_DISABLE_SERVER_GUI", String.valueOf(false)))) // Paper
                 if (flag1 && !GraphicsEnvironment.isHeadless()) {
-                    dedicatedserver1.bd();
+                    //dedicatedserver1.bd();
                 }
 
                 if (optionset.has("port")) {
