From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: jmp <jasonpenilla2@me.com>
Date: Mon, 17 May 2021 16:43:42 -0700
Subject: [PATCH] issue reprod test


diff --git a/src/main/java/com/mojang/datafixers/TypeRewriteRule.java b/src/main/java/com/mojang/datafixers/TypeRewriteRule.java
index 6f836c3382ed83e07000c0a5e4b6e9b69f8c45bf..4c6c38e5fd767202dd81bbc42da058608a6daae8 100644
--- a/src/main/java/com/mojang/datafixers/TypeRewriteRule.java
+++ b/src/main/java/com/mojang/datafixers/TypeRewriteRule.java
@@ -13,6 +13,7 @@ import java.util.Optional;
 import java.util.function.Consumer;
 import java.util.function.Supplier;
 
+// random edit to library source file for issue reproduction
 public interface TypeRewriteRule {
     <A> Optional<RewriteResult<A, ?>> rewrite(final Type<A> type);
 
